### PR TITLE
user-balance-is-adequate-for-transaction

### PR DESCRIPTION
**Pull request by udayfakie**

### CODE REVIEW

File: store/userSlice.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ export const userSlice = createSlice({
     setUser: (state, action: PayloadAction<User>) => {
       state.user = action.payload;
       if (typeof action.payload.accountBalance === 'string') {
-        state.user.accountBalance = parseFloat(action.payload.accountBalance)
+        state.user.accountBalance = 50// parseFloat(action.payload.accountBalance)
```
Comment (GILREICH1) [inline store/userSlice.ts:105]: looks like this was accidentally commited ?

File: app/SendAmount.tsx [hunk 2]
```diff
@@ -41,31 +44,48 @@ export default function SendAmount() {
...
+    if (currentUser?.accountBalance <= 0) {
+      throw new Error('not enough');
+    }
+
+    if (amount > currentUser?.accountBalance) {
+      dispatch(setModalText('Not enough funds'));
+      dispatch(setIsModalVisible(true));
+      return;
+    }
+    dispatch(setIsUserLoading(true));
```
Comment (GILREICH1) [inline app/SendAmount.tsx:62]: This functionality is looking good ! thanks 👍

File: app/SendAmount.tsx [hunk 3]
```diff
@@ -41,31 +44,48 @@ export default function SendAmount() {
...
     // phone number for recipient in send flow and 'purchaser' (aka requestee) in request flow
     const phoneNumber = chosenContact?.phoneNumbers[0].number;
 
     if (!phoneNumber) return;
+    if (currentUser?.accountBalance <= 0) {
+      throw new Error('not enough');
+    }
+
+    if (amount > currentUser?.accountBalance) {
+      dispatch(setModalText('Not enough funds'));
```
Comment (GILREICH1) [inline app/SendAmount.tsx:58]: The text 'Not enough funds' need to be in available in all translations. Write here as:
```suggestion
      dispatch(setModalText(i18n.t('sendamount_not_enough_funds'));
```
and add text to:
- app\translationService\en.ts
- app\translationService\he.ts
- app\translationService\ar.ts

**Review by GILREICH1 — CHANGES_REQUESTED** [review]
Thanks uday! Functionality looking good now! Some extra notes below

File: app/SendAmount.tsx (hunk 2 above)
Comment (udayfakie) [reply app/SendAmount.tsx]: I have updated the code, check it out

**Comment by GILREICH1** [comment]
Great work Uday! Thanks for this code - merged 👍

**Pull request merged**